internal/socket.go

```diff
@@ -70,9 +72,48 @@ func getUnixCreds(conn net.Conn) (*unix.Ucred, error) {
 		return nil, fmt.Errorf("doing rawConn Control: %w", err)
 	}
 
+	if err := authenticateUser(ucred); err != nil {
+		return nil, err
+	}
+
 	return ucred, nil
 }
 
+func authenticateUser(ucred *unix.Ucred) error {
+	// root?
+	if ucred.Uid == 0 {
+		return nil
+	}
+	userInfo, err := user.LookupId(fmt.Sprintf("%d", ucred.Uid))
+	if err != nil {
+		return fmt.Errorf("authenticate user, lookup user info: %s", err)
+	}
+	// user has sudo?
+	// #nosec G204 -- input is properly validated
+	out, err := exec.Command("sudo", "-l", "-U", userInfo.Username).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("authenticate user, check user has sudo: %s", err)
+	}
+	if !strings.Contains(string(out), "not allowed to run sudo") {
+		return nil
+	}
+	// user belongs to nordvpn group?
+	groups, err := userInfo.GroupIds()
+	if err != nil {
+		return fmt.Errorf("authenticate user, check user groups: %s", err)
+	}
+	for groupId := range groups {
```

Comment:
wouldn't it be safer to check if the user is part of `sudo` group here, or is it better with the above check?

Reply:
There are 2 ways to have sudo permissions: 1) via sudo group 2) via sudoers file.
Cannot cover both just with checking group.